### PR TITLE
Copr: Vectorize `Length`

### PR DESCRIPTION
**Pull request by FateTHarlaown**
Signed-off-by: FateTHarlaown <623239185@qq.com>
PCP #5751

###  What have you changed?
Vectorize `Length`

Don't assume reviewers understand the original issue.

###  What is the type of the changes?
- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
no

### CODE REVIEW

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#[cfg(test)]
+mod tests {
+    use tipb::ScalarFuncSig;
+
+    use crate::rpn_expr::types::test_util::RpnFnScalarEvaluator;
+
+    #[test]
+    fn test_length() {
+        let test_cases = vec![
+            (ScalarFuncSig::Length, "", Some(0i64)),
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: Null case is missing

**Comment by breeswish** [comment]
Thanks! LGTM. Please also add corresponding copr-test PR and tidb PR.

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/23

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    use tipb::ScalarFuncSig;
+
+    use super::*;
+    use crate::rpn_expr::types::test_util::RpnFnScalarEvaluator;
+
+    #[test]
+    fn test_length() {
+        let test_cases = vec![
+            (ScalarFuncSig::Length, "", Some(0i64)),
+            (ScalarFuncSig::Length, "你好", Some(6i64)),
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: You may modify the case into:

```rust
let test_cases = vec![
    (None, None),
    (Some(""), Some(0i64)),
    ...
]
```

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
+use tidb_query_codegen::rpn_fn;
+
+use crate::codec::data_type::Bytes;
+use crate::error::Result;
+
+#[rpn_fn]
+#[inline]
+pub fn length(arg: &Option<Bytes>) -> Result<Option<i64>> {
+    Ok(match arg {
+        Some(s) => Some(s.len() as i64),
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: Would you mind use `map` instead?

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    #[test]
+    fn test_length() {
+        let test_cases = vec![
+            (ScalarFuncSig::Length, "", Some(0i64)),
+            (ScalarFuncSig::Length, "你好", Some(6i64)),
+            (ScalarFuncSig::Length, "TiKV", Some(4i64)),
+            (ScalarFuncSig::Length, "あなたのことが好きです", Some(33i64)),
+            (ScalarFuncSig::Length, "분산 데이터베이스", Some(25i64)),
+            (ScalarFuncSig::Length, "россия в мире  кубок", Some(38i64)),
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_string.rs]:
```sql
mysql iosmanthus@(none):test> select length("россия в мире  кубок")                                                               
+--------------------------------+
| length("россия в мире  кубок") |
+--------------------------------+
| 36                             |
+--------------------------------+
1 row in set
Time: 0.024s
```

Comment (breeswish) [reply components/tidb_query/src/rpn_expr/impl_string.rs]: Looks like this is either a GitHub problem or browser problem.. It is fine when you copy the string from local file.

**Comment by FateTHarlaown** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/23 tidb=pr/13374

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+#[cfg(test)]
+mod tests {
+    use tipb::ScalarFuncSig;
+
+    use crate::rpn_expr::types::test_util::RpnFnScalarEvaluator;
+
+    #[test]
+    fn test_length() {
+        let test_cases = vec![
+            (ScalarFuncSig::Length, None, None),
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: Please de-duplicate "ScalarFuncSig::Length"

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 6]
```diff
@@ -0,0 +1,40 @@
+use tidb_query_codegen::rpn_fn;
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: Missing license header

**Comment by breeswish** [comment]
Failed tests are not related, force merge.

**Pull request merged**